RecoEgamma/EgammaElectronProducers/plugins/LowPtGsfElectronIDProducer.cc

```diff
@@ -135,6 +154,29 @@ void LowPtGsfElectronIDProducer::produce(edm::StreamID, edm::Event& event, const
           }
           return ele.electronID(kUnbiased);
         },
+        [&](auto const& ele) {  // trkFunctor ...
+          if (useGsfToTrack_) {
+            const edm::Ptr<pat::PackedCandidate>* ptr1 =
+                ele.hasUserData("ele2packed") ? ele.template userData<edm::Ptr<pat::PackedCandidate>>("ele2packed")
+                                              : nullptr;
+            const edm::Ptr<pat::PackedCandidate>* ptr2 =
+                ele.hasUserData("ele2lost") ? ele.template userData<edm::Ptr<pat::PackedCandidate>>("ele2lost")
+                                            : nullptr;
+            if (ptr1 != nullptr && ptr1->isNonnull() && ptr1->isAvailable() && ptr1->get() != nullptr &&
+                ptr1->get()->bestTrack() != nullptr) {
+              return ptr1->get()->bestTrack();
+            } else if (ptr2 != nullptr && ptr2->isNonnull() && ptr2->isAvailable() && ptr2->get() != nullptr &&
+                       ptr2->get()->bestTrack() != nullptr) {
+              return ptr2->get()->bestTrack();
```

Comment:
the string of checks is a bit elaborate and repeated
```suggestion
            bool hasBestTrack = [] (const PackedPtr* ptr) {
              return ptr != nullptr && ptr->isNonnull() && ptr->isAvailable() && ptr->get() != nullptr &&
                ptr->get()->bestTrack() != nullptr
            };
            if (hasBestTrack(ptr1)) {
              return ptr1->get()->bestTrack();
            } else if (hasBestTrack(ptr2)) {
              return ptr2->get()->bestTrack();
```

Reply:
```
plugins/LowPtGsfElectronIDProducer.cc:166:34: error: 'hasBestTrack' cannot be used as a function
  166 |             if (hasBestTrack(ptr1)) {
      |                                  ^
```
@slava77 ?

Reply:
please try `auto hasBestTrack =` and `;` at the end of return.

Reply:
Done.